daml-foundations/daml-ghc/daml-compiler/src/DA/Service/Daml/Compiler/Impl/Handle.hs

```diff
@@ -176,7 +175,7 @@ buildDar service file mbExposedModules pkgName sdkVersion buildDataFiles dalfInp
       let pkgModuleNames = S.fromList $ map T.unpack $ LF.packageModuleNames pkg
       let missingExposed = S.fromList (fromMaybe [] mbExposedModules) S.\\ pkgModuleNames
       unless (S.null missingExposed) $ do
-          liftIO $ IdeLogger.logSeriousError (ideLogger service) $
+          liftIO $ IdeLogger.logInfo (ideLogger service) $
```

Comment:
Unless I am missing something this means that the error will not be shown by default. The right solution here is probably to not go via the logger at all and instead emit an error diagnostic.

Reply:
I don't think this was ever right - call logError and then `fail` isn't the way to go. I have fixed it for now by just turning it into a `fail` with an error message, which should be no worse than before, but the whole thing is pretty dodgy. I added a FIXME.

Reply:
I don't think this was ever right - call logError and then `fail` isn't the way to go. I have fixed it for now by just turning it into a `error` with an error message, and abort the whole thing, which should be no worse than before, but the whole thing is pretty dodgy. I added a FIXME.

Reply:
@cocreature can you confirm this seems OK temporarily?

Reply:
LGTM :+1: